src/sgp/MouseSystem.h

```diff
@@ -202,14 +206,22 @@ std::function<void(T*, UINT32)> CallbackPrimarySecondary(
 			if (reason & MSYS_CALLBACK_REASON_RBUTTON_UP) {
 				fRightMouseButtonDown = FALSE;
 			}
-			// We need to guard against deletions of the mouse region during callbacks
-			if (r->uiFlags & EXISTS_FLAG && secondaryAction) {
-				secondaryAction(r, reason);
+			if (secondaryAction) {
+				triggeredEventFns.push_back(secondaryAction);
 			}
 		}
 		// We need to guard against deletions of the mouse region during callbacks
-		if (r->uiFlags & EXISTS_FLAG && allEvents) {
-			allEvents(r, reason);
+		if (allEvents) {
+			triggeredEventFns.push_back(allEvents);
+		}
+
+		for (auto&& f : std::move(triggeredEventFns)) {
```

Comment:
I got into the habit of explicitly doing moves where possible in template code, primarily as a potential optimization.
It can also make a difference in some situations, see https://en.cppreference.com/w/cpp/utility/functional/move_only_function/operator() (although for std::function that does not apply).
In this case it is of course not necessary.